File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: int
    filename: str
    file_type: str
    file_size: int
    status: str
    text_length: int
    chunk_count: int
    processing_error: str | None
    upload_time: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/document.ts
export interface Document {
  id: number;
  filename: string;
  file_type: string;
  file_size: number;
  status: string;
  text_length: number;
  chunk_count: number;
  processing_error: string | null;
  upload_time: string;
}

export type UploadResponse = Document;